### ISSUE TITLE
What are the steps to create the `training.cache` file required during training? 

### ISSUE THREAD

**Issue by cruxcode** [ultralytics/yolov5#8658]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Docs on [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data) don't mention anything about creating a `training.cache` file. We have figured out how to create this file but would like to verify that it's the recommended approach. Here's the script we used to create a `labels.cache` file that is read by the yolov5 program when we run `python train.py ...` that produces `training.cache` file.

```
	import os
	import torch
	import numpy as np
	dic = {}

	for i in os.listdir('images'):
   		with open(f'./labels/{i.split(".jpg")[0]}.txt') as annotation:
       	p = []
       	for j in annotation.readlines():
          		arr = list(map(float,j.strip().split()))
          		p.append(arr)
   		dic[f'train\\train\\images\\{i}'] = [np.array(p, dtype=float), (680,460)]

	torch.save(dic,'labels.cache')
```

To understand this code - Our images are stored inside `train/train/images` directory and the size of each image is `(680, 460)`. The labels are in `labels` directory. Code written by teammate [sanskarg348](https://github.com/sanskarg348).

If our understanding is correct, we would like to add these steps to the [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data).

**Comment by glenn-jocher** [comment]
@cruxcode *.cache files contain all labels. These are created automatically and re-created automatically anytime the underlying data changes. There is no need for any manual steps. If you are having any issues with your data or training please submit a bug report with directions to reproduce, thank you!

**Comment by glenn-jocher** [comment]
@cruxcode code is here:
https://github.com/ultralytics/yolov5/blob/b510957650c890dee876146c43dcda1fdfc279d6/utils/dataloaders.py#L460-L469

**Comment by ohjunee** [comment]
@glenn-jocher p is path lists that stores entire training image file, and I wonder how changing only the extension to cache involves mapping label information.
```
cache_path = (p if p.is_file() else Path(self.label_files[0]).parent).with_suffix('.cache') 
```
